### DISCUSSION TITLE
pnpm run serve - doesn't have script to call - help

### DISCUSSION THREAD

**Discussion by monique77** [scalar/scalar#1779] [Q&A]
I`m trying to do builder of Scalar in my local, I already saw all files, directories and I read a lot of your amazing project. But I have a problem, when I execute pnpm run serve, an output message said that don`t have script. So, I saw in the root directory package.json, but in this file has only dev, So, I can execute pnpm run dev, but not serve. I dont know what I need to do. I want to use this for me documentation my apis. After I  will do a pipeline in my azure to do deploy in my AKS.

Please help me, I know it is not a bug, but I didnt see a package.json to call the command pnpm run serve.

oot@monique-martins:/home/monique/Downloads/Scalar/scalar-main# npm run serve
npm ERR! Missing script: "serve"
npm ERR!
npm ERR! To see a list of scripts, run:
npm ERR!   npm run

npm ERR! A complete log of this run can be found in: /root/.npm/_logs/2024-05-18T17_50_23_058Z-debug-0.log
root@monique-martins:/home/monique/Downloads/Scalar/scalar-main# npm run
Lifecycle scripts included in root:
  postinstall
    pnpm -r dev:prepare
  test
    vitest

available via `npm run-script`:
  build
    turbo build
  build:packages
    turbo --filter './packages/**' build
  build:standalone
    pnpm --filter api-reference play-button build:standalone
  bump
    CI=true pnpm run test && pnpm changeset version
  clean
    pnpm clean:nodeModules; pnpm clean:dist; pnpm clean:turbo; pnpm clean:nuxt
  clean:build
    pnpm clean ; pnpm i && pnpm build:packages
  clean:nodeModules
    find . -name node_modules -type d -exec rm -rf {} \;
  clean:nuxt
    find . -name .nuxt -type d -exec rm -rf {} \;
  clean:dist
    find . -name dist -type d -exec rm -rf {} \;
  clean:turbo
    find . -name .turbo -type d -exec rm -rf {} \;
  dev
    turbo run dev --parallel --concurrency=100 --filter './examples/*' --filter=@scalar/nuxt --filter=@scalar/draggable
  @scalar/cli
    pnpm --filter @scalar/cli run @scalar/cli
  format
    pnpm prettier --write .
  format:check
    pnpm prettier --check .
  format:packages
    pnpm syncpack fix-mismatches --types 'prod,dev' || true && pnpm syncpack set-semver-ranges && vite-node scripts/format-package.ts && pnpm install -r
  lint
    pnpm -r lint:check
  lint:fix
    pnpm -r lint:fix
  pack:packages
    ./scripts/pack.sh
  preview:standalone
    pnpm --filter api-reference preview:standalone
  playwright:install
    npx playwright install --with-deps
  test:ci
    CI=1 vitest --silent
  test:e2e
    pnpm --filter playwright test:e2e
  test:e2e:ui
    pnpm --filter playwright test:e2e:ui
  test:e2e:cdn
    pnpm --filter playwright test:e2e cdn
  test:e2e:themes
    pnpm --filter playwright test:e2e themes
  test:e2e:nuxt
    pnpm --filter playwright test:e2e nuxt
  types:check
    pnpm -r types:check
  bootstrap:package
    vite-node scripts/bootstrap-package/bootstrap-package.ts

Log

root@monique-martins:/home/monique/Downloads/Scalar/scalar-main# cat /root/.npm/_logs/2024-05-18T17_50_23_058Z-debug-0.log
0 verbose cli /usr/bin/node /usr/bin/npm
1 info using npm@10.5.2
2 info using node@v20.13.1
3 timing npm:load:whichnode Completed in 1ms
4 timing config:load:defaults Completed in 2ms
5 timing config:load:file:/usr/lib/node_modules/npm/npmrc Completed in 1ms
6 timing config:load:builtin Completed in 1ms
7 timing config:load:cli Completed in 2ms
8 timing config:load:env Completed in 0ms
9 timing config:load:file:/home/monique/Downloads/Scalar/scalar-main/.npmrc Completed in 0ms
10 timing config:load:project Completed in 2ms
11 timing config:load:file:/root/.npmrc Completed in 2ms
12 timing config:load:user Completed in 2ms
13 timing config:load:file:/usr/etc/npmrc Completed in 0ms
14 timing config:load:global Completed in 0ms
15 timing config:load:setEnvs Completed in 1ms
16 timing config:load Completed in 10ms
17 timing npm:load:configload Completed in 10ms
18 timing config:load:flatten Completed in 2ms
19 timing npm:load:mkdirpcache Completed in 0ms
20 timing npm:load:mkdirplogs Completed in 0ms
21 verbose title npm run serve
22 verbose argv "run" "serve"
23 timing npm:load:setTitle Completed in 0ms
24 timing npm:load:display Completed in 1ms
25 verbose logfile logs-max:10 dir:/root/.npm/_logs/2024-05-18T17_50_23_058Z-
26 verbose logfile /root/.npm/_logs/2024-05-18T17_50_23_058Z-debug-0.log
27 timing npm:load:logFile Completed in 11ms
28 timing npm:load:timers Completed in 0ms
29 timing npm:load:configScope Completed in 0ms
30 timing npm:load Completed in 40ms
31 silly logfile start cleaning logs, removing 1 files
32 silly logfile done cleaning log files
33 timing command:run Completed in 18ms
34 verbose stack Error: Missing script: "serve"
34 verbose stack
34 verbose stack To see a list of scripts, run:
34 verbose stack   npm run
34 verbose stack     at RunScript.run (/usr/lib/node_modules/npm/lib/commands/run-script.js:94:13)
34 verbose stack     at async module.exports (/usr/lib/node_modules/npm/lib/cli-entry.js:61:5)
35 verbose cwd /home/monique/Downloads/Scalar/scalar-main
36 verbose Linux 6.5.0-35-generic
37 verbose node v20.13.1
38 verbose npm  v10.5.2
39 error Missing script: "serve"
39 error
39 error To see a list of scripts, run:
39 error   npm run
40 verbose exit 1
41 timing npm Completed in 80ms
42 verbose code 1
43 error A complete log of this run can be found in: /root/.npm/_logs/2024-05-18T17_50_23_058Z-debug-0.log

**Comment by marclave** [comment]
hi @monique77 ! appreciate your message, to run the examples for scalar you do the following command at root:
```
pnpm dev
```

are you wanting to just document your openapi spec?

we also have a cli to document also https://github.com/scalar/scalar/tree/main/packages/cli#scalar-cli

```
scalar reference openapi.json
```

to render an openapi spec :)

**Comment by monique77** [comment]
I'm glad for your  answer. So, pnpm dev work, but I thought that your project we could use it in mode prod. Let me know if in mode prod needs pay?

Yahoo Mail: Search, Organize, Conquer 
 
  On Sat, May 18, 2024 at 20:33, Marc ***@***.***> wrote:   


hi @monique77 ! appreciate your message, to run the examples for scalar you do the following command at root:
pnpm dev

are you wanting to just document your openapi spec?

we also have a cli to document also https://github.com/scalar/scalar/tree/main/packages/cli#scalar-cli
scalar reference openapi.json

to render an openapi spec :)
